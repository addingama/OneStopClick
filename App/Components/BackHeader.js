import React, { Component, PropTypes } from 'react'
import { Header, Icon } from 'react-native-elements'
import styles from './Styles/DrawerHeaderStyle'

export default class BackHeader extends Component {
  openMenu () {
    this.props.navigation.navigate('DrawerOpen')
  }

  render () {
    const { title, navigation, backAction } = this.props
    return (
      <Header
        backgroundColor='#2F1F37'
        leftComponent={<Icon
          name='ios-arrow-back'
          type='ionicon'
          color='white'
          onPress={() => backAction === null ? navigation.goBack() : backAction()} />}
        centerComponent={{ text: title, style: styles.titleStyle }}
      />
    )
  }
}

BackHeader.propTypes = {
  title: PropTypes.string.isRequired,
  navigation: PropTypes.object.isRequired
}
