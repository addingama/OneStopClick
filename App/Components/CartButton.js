import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { View, Text, TouchableWithoutFeedback } from 'react-native'
import { Icon } from 'react-native-elements'
import styles from './Styles/CartButtonStyle'

export default class CartButton extends Component {
  constructor (props) {
    super(props)
    this.state = {
      BadgeCount: 0
    }
  }

  componentWillReceiveProps (newProps) {
    this.setState({
      BadgeCount: newProps.BadgeCount
    })
  }

  renderBadge (hasBadge) {
    if (hasBadge) {
      return (
        <View style={styles.badgeContainer}>
          <Text style={styles.badgeCount}>
            {this.state.BadgeCount}
          </Text>
        </View>
      )
    }
  }

  render () {
    const { navigation } = this.props

    return (
      <View>
        <TouchableWithoutFeedback>
          <View style={{ flexDirection: 'row' }}>
            <Icon name='shopping-cart' color='white'
              onPress={() => navigation.navigate('CartDetailScreen')}
            />
            { this.renderBadge(this.state.BadgeCount !== 0)}
          </View>
        </TouchableWithoutFeedback>
      </View>
    )
  }
}
