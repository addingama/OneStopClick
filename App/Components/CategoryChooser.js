import React, { Component, PropTypes } from 'react'
import { List, ListItem } from 'react-native-elements'

export default class CategoryChooser extends Component {
  generateItems (items) {
    const { selectedValue, onValueChange } = this.props
    return items.map(function (item) {
      return (
        <ListItem
          hideChevron={selectedValue !== item.name}
          key={item.id}
          title={item.name}
          onPress={() => onValueChange(item.name)}
          rightIcon={{name: 'check-circle', color: 'green'}}
        />
        // <Picker.Item label={item.name} value={item.name} key={item.id} />
      )
    })
  }
  render () {
    const { items } = this.props
    return (
      <List>
        { this.generateItems(items) }
      </List>
    )
  }
}

CategoryChooser.propTypes = {
  selectedValue: PropTypes.string.isRequired,
  items: PropTypes.array.isRequired,
  onValueChange: PropTypes.func.isRequired
}
