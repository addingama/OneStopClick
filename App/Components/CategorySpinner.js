import React, { Component, PropTypes } from 'react'
import { Picker } from 'react-native'
import styles from './Styles/CategorySpinnerStyle'

export default class CategorySpinner extends Component {
  generatePickerItems (items) {
    return items.map(function (item) {
      return (
        <Picker.Item label={item.name} value={item.name} key={item.id} />
      )
    })
  }
  render () {
    const { selectedValue, items, onValueChange } = this.props
    return (
      <Picker selectedValue={selectedValue} onValueChange={(value) => onValueChange(value)}>
        {this.generatePickerItems(items)}
      </Picker>
    )
  }
}

CategorySpinner.propTypes = {
  selectedValue: PropTypes.string.isRequired,
  items: PropTypes.array.isRequired,
  onValueChange: PropTypes.func.isRequired
}
