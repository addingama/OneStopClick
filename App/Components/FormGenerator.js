import { FormLabel, FormInput, FormValidationMessage, Card, Button, Icon, Button as RneButton } from 'react-native-elements'
import I18n from 'react-native-i18n'
import React, { Component } from 'react'
import { View, TouchableOpacity, Text, Alert, FlatList } from 'react-native'
import { validateField } from '../Lib/validator'
import styles from './Styles/FormGeneratorStyle'
import { currency } from '../Lib/numberFormatter.js'
var uuid = require('react-native-uuid')

export class CustomInputField extends Component {
  setValue (fieldName, newValue, existingState) {
    existingState[fieldName].value = newValue
    return existingState
  };

  render () {
    var field = this.props.field
    const { name, secureText, message, returnKeyType, keyboardType, value } = field
    var state = this.props.state
    return <View>
      <FormLabel>{I18n.t(name)}</FormLabel>
      <FormInput
        ref={name}
        value={value}
        secureTextEntry={secureText}
        returnKeyType={returnKeyType}
        keyboardType={keyboardType}
        autoCapitalize='none'
        autoCorrect={false}
        onChangeText={(text) => this.props.updateState(this.setValue(name, text, state))}
        onSubmitEditing={() => this.props.updateState(validateField(state, name))}
        onBlur={() => this.props.updateState(validateField(state, name))}
        {...this.props}
      />
      <FormValidationMessage>{message}</FormValidationMessage>
    </View>
  }
}

export class CustomButton extends React.Component {
  render () {
    return <Button
      {...this.props}
      backgroundColor='#86b200'
      fontWeight='bold'
    />
  }
}

export class HamburgerMenu extends React.Component {
  render () {
    return (
      <TouchableOpacity
        {...this.props}
        underlayColor='transparent'>
        <Icon
          color='white'
          name='menu'
          size={28}
        />
      </TouchableOpacity>
    )
  }
}

export class Category extends Component {
  render () {
    const { name } = this.props.category
    return (<Text style={[styles.titleLabel]}>{name}</Text>)
  }
}

export class Products extends Component {
  addCartItem (product) {
    const { cartItems, historyItems, accessToken } = this.props

    if (accessToken === null || accessToken === '') {
      Alert.alert('Error', 'Please login.')
    } else {
      // checking duplication
      var hasAdded = false
      var hasBought = false

      // on cart
      for (var i = 0; i < cartItems.length; i++) {
        if (cartItems[i].product_id === product.id) {
          hasAdded = true
          break
        }
      }

      // on history
      for (var j = 0; j < historyItems.length; j++) {
        if (historyItems[j].id === product.id) {
          hasBought = true
          break
        }
      }

      if (!hasAdded && !hasBought) {
        var newCartItems = Object.assign([], cartItems)
        newCartItems.push(product)
        this.props.onBuyPress(newCartItems)
      } else {
        if (hasAdded) {
          Alert.alert('Warning', 'You have already added ' + product.product_name + '.')
        }

        if (hasBought) {
          Alert.alert('Warning', 'You have already bought ' + product.product_name + '. Please download from history transaction.')
        }
      }
    }
  }

  render () {
    const { data } = this.props
    var products = Object.assign([], data)
    for (let j = 0; j < products.length; j++) {
      var product = Object.assign({}, products[j])
      product.key = product.id
      products[j] = product
    }

    return (<FlatList
      data={products}
      key={uuid.v1()}
      numColumns='2'
      renderItem={({ item }) =>
        <TouchableOpacity onPress={() => this.props.onProductClick(item)}>
          <View>
            <Card
              style={styles.cardContent}
              key={item.id}
              title={item.product_name}
              image={{ uri: item.images[0].image_url }}
            >
              <Text
                numberOfLines={2}
                ellipsizeMode={'tail'}
                style={{ marginBottom: 10 }}>
                {item.description}
              </Text>
              <Text
                style={{ marginBottom: 10, color: 'green' }}>
                Rp. {currency(item.price)}
              </Text>
              <RneButton
                icon={{ name: 'shopping-cart' }}
                backgroundColor='green'
                fontFamily='Lato'
                style={{ width: 120, margin: 0, padding: 0 }}
                onPress={() => this.addCartItem(item)}
                title={I18n.t('addToCart')} />
            </Card>
          </View>
        </TouchableOpacity>
      }
    />)
  }
}
