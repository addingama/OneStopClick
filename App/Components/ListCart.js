import React, { Component } from 'react'
import { Text, View, Image, TouchableHighlight, FlatList, TouchableWithoutFeedback } from 'react-native'
import styles from './Styles/ListCartStyle'
import { currency } from '../Lib/numberFormatter.js'
var uuid = require('react-native-uuid')

class ListCart extends Component {
  render () {
    const { cartItems, accessToken } = this.props

    return (
      <FlatList style={styles.container}
        data={cartItems}
        key={cartItems.id}
        numColumns='1'
        renderItem={({ item }) =>
        /* Adding touch event to activated scrolling
        <Image
                  style={styles.image}
                  source={{ uri: item.product.images[0].image_url }}
            /> */
          <TouchableWithoutFeedback>
            <View>
              <View style={styles.rowContainer}>

                <View style={styles.textContainer}>
                  <View>
                    <Text
                      numberOfLines={2}
                      ellipsizeMode={'tail'}
                      style={styles.productTitle}>
                      {item.product.product_name}
                    </Text>
                  </View>
                  <View>
                    <Text
                      numberOfLines={2}
                      ellipsizeMode={'tail'}
                      style={styles.productDesc}>
                      {item.product.description}
                    </Text>
                  </View>
                  <View>
                    <Text
                      style={styles.productPrice}>
                    Rp. { currency(item.product.price) }
                    </Text>
                  </View>
                </View>
                <View style={styles.deleteContainer}>
                  <TouchableHighlight onPress={() => {
                    this.props.removeCartItem(accessToken, item)
                  }}>
                    <Text style={styles.delete}>X</Text>
                  </TouchableHighlight>
                </View>
              </View>
              <View style={styles.separator} />
            </View>
          </TouchableWithoutFeedback>
        }
      />
    )
  }
}

export default ListCart
