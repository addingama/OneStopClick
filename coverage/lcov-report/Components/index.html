<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> Components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.24% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>43/68</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">34.21% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>13/38</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>13/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.18% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>43/67</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="BackHeader.js"><a href="BackHeader.js.html">BackHeader.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file high" data-value="CategoryChooser.js"><a href="CategoryChooser.js.html">CategoryChooser.js</a></td>
	<td data-value="85.71" class="pic high"><div class="chart"><div class="cover-fill" style="width: 85%;"></div><div class="cover-empty" style="width:15%;"></div></div></td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	</tr>

<tr>
	<td class="file high" data-value="Col.js"><a href="Col.js.html">Col.js</a></td>
	<td data-value="87.5" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="45.45" class="pct low">45.45%</td>
	<td data-value="11" class="abs low">5/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	</tr>

<tr>
	<td class="file medium" data-value="DrawerHeader.js"><a href="DrawerHeader.js.html">DrawerHeader.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="FormGenerator.js"><a href="FormGenerator.js.html">FormGenerator.js</a></td>
	<td data-value="29.17" class="pic low"><div class="chart"><div class="cover-fill" style="width: 29%;"></div><div class="cover-empty" style="width:71%;"></div></div></td>
	<td data-value="29.17" class="pct low">29.17%</td>
	<td data-value="24" class="abs low">7/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="12" class="abs low">3/12</td>
	<td data-value="30.43" class="pct low">30.43%</td>
	<td data-value="23" class="abs low">7/23</td>
	</tr>

<tr>
	<td class="file medium" data-value="Grid.js"><a href="Grid.js.html">Grid.js</a></td>
	<td data-value="72.73" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 72%;"></div><div class="cover-empty" style="width:28%;"></div></div></td>
	<td data-value="72.73" class="pct medium">72.73%</td>
	<td data-value="11" class="abs medium">8/11</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="12" class="abs low">3/12</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="72.73" class="pct medium">72.73%</td>
	<td data-value="11" class="abs medium">8/11</td>
	</tr>

<tr>
	<td class="file high" data-value="ProgressIndicator.js"><a href="ProgressIndicator.js.html">ProgressIndicator.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="Row.js"><a href="Row.js.html">Row.js</a></td>
	<td data-value="87.5" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="45.45" class="pct low">45.45%</td>
	<td data-value="11" class="abs low">5/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Aug 30 2017 11:23:09 GMT+0700 (WIB)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
