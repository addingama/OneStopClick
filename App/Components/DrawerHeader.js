import React, { Component, PropTypes } from 'react'
import { Header } from 'react-native-elements'
import { HamburgerMenu } from './FormGenerator'
import styles from './Styles/DrawerHeaderStyle'

export default class DrawerHeader extends Component {
  openMenu () {
    this.props.navigation.navigate('DrawerOpen')
  }

  render () {
    const { title, rightComponent } = this.props
    return (
      <Header
        backgroundColor='#2F1F37'
        leftComponent={<HamburgerMenu onPress={this.openMenu.bind(this)} />}
        centerComponent={{ text: title, style: styles.titleStyle }}
        rightComponent={rightComponent}
      />
    )
  }
}

DrawerHeader.propTypes = {
  title: PropTypes.string.isRequired,
  navigation: PropTypes.object.isRequired
}
