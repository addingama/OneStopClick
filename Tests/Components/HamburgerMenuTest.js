import 'react-native'
import React from 'react'
import { HamburgerMenu } from '../../App/Components/FormGenerator'
import renderer from 'react-test-renderer'

test('HamburgerMenu component renders correctly', () => {
  const tree = renderer.create(<HamburgerMenu />).toJSON()
  expect(tree).toMatchSnapshot()
})
