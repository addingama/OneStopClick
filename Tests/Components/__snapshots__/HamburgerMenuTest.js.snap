// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    onPress={undefined}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    underlayColor="white"
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 28,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î—’
    </Text>
  </View>
</View>
`;
